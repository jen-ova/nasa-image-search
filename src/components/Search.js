import React, { useState } from "react";
import "../styles/Search.css";

function Search() {
  const [value, setValue] = useState("");

  return (
    <>
      <form>
        <input
          className="search-input"
          type="text"
          onChange={(event) => setValue(event.target.value)}
          value={value}
        />
        <button type="submit" className="search-button">
          Search
        </button>
      </form>
    </>
  );
}

export default Search;
