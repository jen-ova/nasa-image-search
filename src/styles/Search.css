.search-form {
    display: flex;
    flex-direction: row;
}

.search-input {
    width: 20rem;
    height: 2.5rem;
    border-radius: 1rem;
    font-size: 1rem;
    padding: 0 1rem;
    margin-right: 1rem;
}

.search-button {
    width: 5rem;
    border-radius: 1rem;
    background-color: #be0317;
    color: white;
    font-weight: bold;
    font-size: 1rem;
}

