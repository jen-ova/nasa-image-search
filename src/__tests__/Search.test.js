import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

it("renders correctly", () => {
  const { asFragment } = render(<Search />);

  expect(asFragment()).toMatchSnapshot();
});
