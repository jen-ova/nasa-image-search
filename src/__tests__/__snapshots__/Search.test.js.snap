// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <form>
    <input
      class="search-input"
      type="text"
      value=""
    />
    <button
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
